src/raftstore/store/peer.rs

```diff
@@ -1136,7 +1138,11 @@ impl Peer {
                 if lease_to_be_updated {
                     let propose_time = self.find_propose_time(entry.get_index(), entry.get_term());
                     if let Some(propose_time) = propose_time {
-                        self.maybe_renew_leader_lease(&ctx.local_reader, propose_time);
+                        if let Some(progress) = self.maybe_renew_leader_lease(propose_time) {
+                            let mut meta = ctx.store_meta.lock().unwrap();
+                            let reader = meta.readers.get_mut(&self.region_id).unwrap();
+                            self.maybe_update_read_progress(reader, progress);
```

Comment:
Please call it in `maybe_renew_leader_lease` instead of writing this form every time.